python/simpler/worker.py

```
@@ -1434,165 +1503,159 @@ def _run_chip_main_loop(  # noqa: PLR0912, PLR0913, PLR0915 -- unified TASK_READ
+                raise RuntimeError(
+                    f"chip_process dev={device_id}: cid {cid} not prepared before TASK_READY "
+                    f"(register via _CTRL_PREPARE first)"
+                )
+            # Redirect any registered host pointer (a parent VA) in the
+            # blob to this child's own mapping before the runtime reads it.
+            # No-op when nothing is registered.
+            if host_buf_ranges:
+                _rewrite_blob_host_addrs(buf, _OFF_TASK_ARGS_BLOB, host_buf_ranges)
+            # Hand the mailbox bytes straight to C++ (zero-copy zero-decode):
+            # the blob layout is what `write_blob` already wrote, so re-parsing
+            # it in Python is N×40B of avoidable work and a permanent
```

Ruff (0.15.21)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)